examples/tests/valid/resource.w

```diff
@@ -0,0 +1,7 @@
+resource R {
+  name: str;
+  
+  init(name: str) {
+    this.name = name;
+  }
+}
```

Comment:
Maybe add something like this to ensure it's running the constructor as expected

```suggestion
resource R {
  name: str;
  
  init(name: str) {
    this.name = name;
  }
}

let res = new R("Arr");
print(resource.name);
```